grid_generation/Gen_CICE_grid.py

```diff
@@ -0,0 +1,167 @@
+       
+    # Area
+    AREA  = (in_superGridFile['area'])
+    TAREA = AREA[::2,::2] + AREA[1::2,1::2] + AREA[::2,1::2] +  AREA[1::2,::2]
+    
+    # UAREA need to wrap around the globe. Copy ocn_area and
+    # add an extra column at the end. Also u-cells cross the
+    # tri-polar fold so add an extra row at the top.
+    area_ext = np.append(AREA[:], AREA[:, 0:1], axis=1)
+    area_ext = np.append(area_ext[:], area_ext[-1:, :], axis=0)
+
+    UAREA = area_ext[1::2, 1::2] + area_ext[2::2, 1::2] + \
+                area_ext[2::2, 2::2] + area_ext[1::2, 2::2]
+
+    # Close input files
+    in_superGridFile.close()
+
+    # Create a new NetCDF file
+    nc = Dataset(output_file, 'w', format='NETCDF4')
+
+    # Define dimensions
+    ny, nx = ULAT.shape
+    nc.createDimension('ny', ny)
+    nc.createDimension('nx', nx)
+
+    # Define variables
+    ulat = nc.createVariable('ulat', 'f8', ('ny', 'nx'), compression='zlib', complevel=1)
+    ulon = nc.createVariable('ulon', 'f8', ('ny', 'nx'), compression='zlib', complevel=1)
+    tlat = nc.createVariable('tlat', 'f8', ('ny', 'nx'), compression='zlib', complevel=1)
+    tlon = nc.createVariable('tlon', 'f8', ('ny', 'nx'), compression='zlib', complevel=1)
+    htn = nc.createVariable('htn', 'f8', ('ny', 'nx'), compression='zlib', complevel=1)
+    hte = nc.createVariable('hte', 'f8', ('ny', 'nx'), compression='zlib', complevel=1)
+    angle = nc.createVariable('angle', 'f8', ('ny', 'nx'), compression='zlib', complevel=1)
+    angleT = nc.createVariable('angleT', 'f8', ('ny', 'nx'), compression='zlib', complevel=1)
+    tarea = nc.createVariable('tarea', 'f8', ('ny', 'nx'), compression='zlib', complevel=1)
+    uarea = nc.createVariable('uarea', 'f8', ('ny', 'nx'), compression='zlib', complevel=1)
+
+    # Add attributes
+    ulat.units = "radians"
+    ulat.title = "Latitude of U points"
+    ulon.units = "radians"
+    ulon.title = "Longitude of U points"
+    tlat.units = "radians"
+    tlat.title = "Latitude of T points"
+    tlon.units = "radians"
+    tlon.title = "Longitude of T points"
+    htn.units = "cm"
+    htn.title = "Width of T cells on N side."
+    hte.units = "cm"
+    hte.title = "Width of T cells on E side."
+    angle.units = "radians"
+    angle.title = "Rotation angle of U cells."
+    angleT.units = "radians" 
+    angleT.title = "Rotation angle of T cells." 
+    tarea.units = "m^2" 
+    tarea.title = "Area of T cells." 
+    uarea.units = "m^2" 
+    uarea.title = "Area of U cells." 
+
+    # Add versioning information    
+    if is_git_repo():
+        git_url, git_hash, rel_path = git_info()
+        nc.inputfile = f"{in_superGridPath}"
+        nc.timeGenerated = f"{datetime.now()}"
+        nc.created_by = f"{os.environ.get('USER')}"
+        nc.history = f"Created using commit {git_hash} of {git_url}"
+    else:
+        nc.inputfile = f"{in_superGridPath}"
+        nc.timeGenerated = f"{datetime.now()}"
+        nc.created_by = f"{os.environ.get('USER')}"
```

Comment:
```
nc.inputfile = f"{in_superGridPath}"
nc.timeGenerated = f"{datetime.now()}"
nc.created_by = f"{os.environ.get('USER')}"
```

These three lines can go outside the if/else as they are the same in both cases.

Can we add a hash for the input file? Similar to the payu manifests md5 hash?